src/network/networkd-dhcp6.c

```diff
@@ -35,6 +36,88 @@ static bool dhcp6_get_prefix_delegation(Link *link) {
                       RADV_PREFIX_DELEGATION_BOTH);
 }
 
+static bool dhcp6_has_preferred_subnet_id(Link *link) {
+        assert(link->network);
+        return link->network->router_prefix_subnet_id >= 0;
+}
+
+static int dhcp6_get_preferred_delegated_prefix(
+                Manager* manager,
+                Link *link,
+                const struct in6_addr *pd_prefix,
+                uint8_t pd_prefix_len,
+                struct in6_addr *ret_addr) {
+        int r;
+        union in_addr_union pd_prefix_union = { .in6 = *pd_prefix, };
```

Comment:
why the `,`?

Reply:
It has become customary to me to just end lists of (i.e. initializers) with a `,`. I'll remove this in the next revision.

Reply:
so, actually we do that too. But we generally line break structure initializations. And when we do that, the `,` isn't so weird anymore. i.e. consider changing this to:

```c
union in_addr_union pd_prefix_union = {
        .in6 = *pd_prefix,
};
```

so much nicer!

(here and for all other structure initializations)